pkg/secrets/secrets.go

```diff
@@ -47,11 +47,34 @@ func Init(command string, arguments []string, timeout int, maxSize int) {
 
 type walkerCallback func(string) (string, error)
 
+// Viper support setting chunk of configuration through env variable using
+// Yaml/json. Sadly those are loaded on the fly when querying the
+// configuration, the underlying type remain a string. In order to support
+// nested `ENC` in JSON payload we try to unmarshal each string. This is costly
+// but only done once when loading the configuration when the agent starts.
+func handleString(str string, callback walkerCallback) (interface{}, error) {
+	var data interface{}
+	err := yaml.Unmarshal([]byte(str), &data)
```

Comment:
Should we handle this error somewhere?

Reply:
If we fail to load the yaml payload we can discard the error and handle it as a raw string as normal.